### PR TITLE
Create endpoint and formatter for gather cms data

### PR DESCRIPTION
**Pull request by joyet-simon**
### Reason for change

[Linear task](https://linear.app/almapay/issue/ECOM-2116/[-php-client]-create-endpoint-that-returns-data)
We need to implement new endpoint to save an url for gathering CMS data. 

### Code changes

- Add new endpoint in Merchants.php file
- Add a formatter for gathering CMS data payload
- Add CmsInfos and CmsFeatures Entity
- Add some unit tests

### How to test

_As a reviewer, you are encouraged to test the PR locally._

You can run unit tests

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: tests/Unit/Entities/CmsFeaturesTest.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		$this->assertFalse($cmsFeatures->getProperties()['log_activated']);
+		$this->assertEquals(['category1', 'category2'], $cmsFeatures->getProperties()['excluded_categories']);
+		$this->assertTrue($cmsFeatures->getProperties()['excluded_categories_activated']);
+		$this->assertEquals(['feature1', 'feature2'], $cmsFeatures->getProperties()['specific_features']);
+		$this->assertEquals(['FR', 'US'], $cmsFeatures->getProperties()['country_restriction']);
+		$this->assertFalse($cmsFeatures->getProperties()['is_multisite']);
+		$this->assertTrue($cmsFeatures->getProperties()['custom_widget_css']);
+	}
+
+	public function testConstructorHandlesNullValuesCorrectly()
```
Comment (Francois-Gomis) [inline tests/Unit/Entities/CmsFeaturesTest.php:45]: add new test with  empty array

File: tests/Unit/Entities/CmsInfoTest.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		$this->assertArrayNotHasKey('language_name', $cmsInfo->getProperties());
+		$this->assertArrayNotHasKey('language_version', $cmsInfo->getProperties());
+		$this->assertArrayNotHasKey('alma_plugin_version', $cmsInfo->getProperties());
+		$this->assertArrayNotHasKey('alma_sdk_version', $cmsInfo->getProperties());
+		$this->assertArrayNotHasKey('alma_sdk_name', $cmsInfo->getProperties());
+		$this->assertEquals([], $cmsInfo->getProperties()['third_parties_plugins']);
+		$this->assertEquals([], $cmsInfo->getProperties()['themes']);
+	}
+
+	public function testGetPropertiesFiltersOutNullAndEmptyValues()
```
Comment (Francois-Gomis) [inline tests/Unit/Entities/CmsInfoTest.php:67]: add test with empty array

**Review by Francois-Gomis — CHANGES_REQUESTED** [review]

File: src/Lib/PayloadFormatter.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+use Alma\API\Entities\MerchantData\CmsInfo;
+
+class PayloadFormatter
+{
+	/**
+	 * @param CmsInfo $cmsInfo
+	 * @param CmsFeatures $cmsFeatures
+	 * @return string
+	 */
+	public static function formatIntegrationConfigurationPayload(CmsInfo $cmsInfo, CmsFeatures $cmsFeatures)
```
Comment (Benjamin-Freoua-Alma) [inline src/Lib/PayloadFormatter.php:15]: We don't need static function here

Comment (Benjamin-Freoua-Alma) [inline src/Lib/PayloadFormatter.php:15]: Maybe rename the function without `integration` WDYT ?

File: tests/Unit/Endpoints/MerchantsTest.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+use Mockery;
+use PHPUnit\Framework\TestCase;
+
+class MerchantsTest extends TestCase
+{
+	const URL = "https://www.example.com/integrations/configurations";
+
+	public function setUp(): void
+	{
+		$this->merchantEndpoint = Mockery::mock(Merchants::class)->makePartial();
```
Comment (Benjamin-Freoua-Alma) [inline tests/Unit/Endpoints/MerchantsTest.php:18]: We don't want to remove Mockery and use native mock ?

**Review by Benjamin-Freoua-Alma — APPROVED** [review]

**Review by Benjamin-Freoua-Alma — COMMENTED** [review]
change the review approvement

**Review by Benjamin-Freoua-Alma — COMMENTED** [review]
Some comments has been see in mob review :)

File: tests/Unit/Endpoints/MerchantsTest.php (hunk 4 above)
Comment (joyet-simon) [reply tests/Unit/Endpoints/MerchantsTest.php:18]: When we want to test an endpoint, I'm note sure we can do it without a makePartial

**Review by Francois-Gomis — APPROVED** [review]

File: tests/Unit/Endpoints/MerchantsTest.php (hunk 4 above)
Comment (Benjamin-Freoua-Alma) [reply tests/Unit/Endpoints/MerchantsTest.php:18]: I saw their is a similar like Mockery with a native PhpUnit, but we can explore it for a next PR :)

**Review by Benjamin-Freoua-Alma — APPROVED** [review]

**Review by Francois-Gomis — APPROVED** [review]

**Review by Benjamin-Freoua-Alma — APPROVED** [review]
